src/librustc/infer/mod.rs

```diff
@@ -479,6 +480,7 @@ impl<'a, 'gcx, 'tcx> TyCtxt<'a, 'gcx, 'gcx> {
         InferCtxtBuilder {
             global_tcx: self,
             arena: SyncDroplessArena::default(),
+            interners: None,
```

Comment:
I'm curious -- why move the initialization of `interners` into `enter_local`?  Why not just create it here and not make it an `Option`?

Also, do we care that if you invoke `enter` twice it will get overwritten? I guess that's not what you're supposed to do here.

Reply:
I didn't move initialization of `CtxtInterners` into `enter_local`. It was already initialized there. However it needs to live as long as the arena, so I created storage for it along side `arena`. You can think of `Option<CtxtInterners<'tcx>>` as a cheap `TypedArena<CtxtInterners<'tcx>>` here.

I could add an assertion to avoid overwriting the `Option`s used here.